### PR TITLE
RUMM-594 Provide integration with SQLDelight

### PR DESCRIPTION
**Pull request by mariusc83**
### What does this PR do?

The scope of this PR is to provide integration helpers with the [SQLDelight API](https://cashapp.github.io/sqldelight/android_sqlite/).

First commit adds a use case of the **SQLDelight** Api into the kotlin sample application.

In the second commit we introduce a new module/artifact : **dd-sdk-android-sqldelight**, where we provide a an extension of the `AndroidSqliteDriver.Callback` class : `DatadogSqliteCallback`. By doing so we're going to be able to intercept the `onCorruption` callback and send relevant RUM error events for each database corruption case.

### Motivation

What inspired you to submit this pull request?

### Additional Notes

Anything else we should know when reviewing?

### Review checklist (to be filled by reviewers)

- [ ] Feature or bugfix MUST have appropriate tests (unit, integration, e2e)
- [ ] Make sure you discussed the feature or bugfix with the maintaining team in an Issue
- [ ] Make sure each commit and the PR mention the Issue number (cf the [CONTRIBUTING](CONTRIBUTING.md) doc)

### CODE REVIEW

File: sample/kotlin/src/main/sqldelight/com.datadog.android.sample.data.db/Logs.sq [hunk 1]
```diff
@@ -0,0 +1,17 @@
+CREATE TABLE IF NOT EXISTS logs (
```
Comment (xgouchet) [inline sample/kotlin/src/main/sqldelight/com.datadog.android.sample.data.db/Logs.sq:1]: Is this file supposed to be in the `main` source set ? Is it linked with SQLDelight?

**Review by xgouchet — APPROVED** [review]

File: sample/kotlin/src/main/sqldelight/com.datadog.android.sample.data.db/Logs.sq (hunk 1 above)
Comment (mariusc83) [reply sample/kotlin/src/main/sqldelight/com.datadog.android.sample.data.db/Logs.sq:1]: yeah...they specifically wanted there, I tried to move it in the flavour source folder but then at compile time they are not able to find the generated class. I might try a different trick there actually but if that doesn't work I am afraid I will have to keep it in the main.

File: dd-sdk-android-sqldelight/README.md [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+}
+```
+
+Following SQLDelight's [Generated API documentation][1], you just have to provide the `DatadogSqliteCallback` in the 
+`AndroidSqliteDriver` constructor.
+
+Doing this detects whenever a database is corrupted and sends a relevant
+RUM error event for it.
+
+Kotlin:
```
Comment (xgouchet) [inline dd-sdk-android-sqldelight/README.md:45]: You probably don't need to say this :/ We don't say it anywhere else so it's a bit confusing.

**Review by xgouchet — APPROVED** [review]

File: dd-sdk-android-sqldelight/README.md (hunk 2 above)
Comment (mariusc83) [reply dd-sdk-android-sqldelight/README.md:45]: We are mentioning it also for Sqlite here : https://github.com/DataDog/dd-sdk-android/blob/master/docs/rum_collection.md. 
I really think we should tell the users what we are trying to do with these callbacks (this was actually corrected by the docs team, I am not sure what was the original phrase I added there).

Oh...I just realised that I added the [...] without the // there , will fix it now.

**Pull request merged**